### DISCUSSION TITLE
Detect errors in the model training process   Please！Rookie

### DISCUSSION THREAD

**Discussion by 18217186223** [PaddlePaddle/PaddleOCR#13882] [Q&A]
python tools/train.py -c pretrain_models/ch_PP-OCRv4_det_cml.yml
Traceback (most recent call last):
  File "D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\tools\train.py", line 252, in <module>
    config, device, logger, vdl_writer = program.preprocess(is_train=True)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\tools\program.py", line 717, in preprocess
    config = merge_config(config, profile_dic)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\tools\program.py", line 97, in merge_config
    config[key] = value
    ~~~~~~^^^^^
TypeError: 'str' object does not support item assignment


code
D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\tools\program.py
error code
def merge_config(config, opts):
    """
    Merge config into global config.
    Args:
        config (dict): Config to be merged.
    Returns: global config
    """
    for key, value in opts.items():
        if "." not in key:
            if isinstance(value, dict) and key in config:
                config[key].update(value)
            else:
                config[key] = value

set
Global:","  debug:false","  use_gpu:false","  epoch_num:500","  log_smooth_window:20","  print_batch_step:20","  save_model_dir:./output/ch_PP-OCRv4","  save_epoch_step:50","  eval_batch_step:","  - 0","  - 1000","  cal_metric_during_train:true","  checkpoints:null","  pretrained_model:D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\pretrain_models\ch_PP-OCRv4_det_train\ch_PP-OCRv4_det_train\best_accuracy","  save_inference_dir:null","  use_visualdl:false","  infer_img:doc/imgs_en/img_10.jpg",
Train:","  dataset:","    name:SimpleDataSet","    data_dir:D:\TOOLSSS\PYFILE\test_data\train","    label_file_list:D:\TOOLSSS\py3.11\Lib\site-packages\train_data\det\train.txt","      - ./train_data/icdar2015/text_localization/train_icdar2015_label.txt"
Eval:","  dataset:","    name:SimpleDataSet","    data_dir:D:\TOOLSSS\PYFILE\test_data\train","    label_file_list:D:\TOOLSSS\py3.11\Lib\site-packages\train_data\det\val.txt","      - ./train_data/icdar2015/text_localization/test_icdar2015_label.txt"

If you need additional code or settings, please let me know, I don't quite know what need to fix the problem
THKS THKS!!!

**Comment by UserWangZz** [comment]
TypeError: 'str' object does not support item assignment
Maybe your config file have some error?

**Reply by 18217186223** [reply]
The configuration file is a ch_PP-OCRv4_det_cml.yml downloaded from the official website
Only the above modifications have been made to the Global，Train，Eval：pretrained_model:D:\TOOLSSS\py3.11\Lib\site-packages\paddleocr\pretrain_models\ch_PP-OCRv4_det_train\ch_PP-OCRv4_det_train\best_accuracy  ；data_dir:D:\TOOLSSS\PYFILE\test_data\train"," label_file_list:D:\TOOLSSS\py3.11\Lib\site-packages\train_data\det\train.txt"；data_dir:D:\TOOLSSS\PYFILE\test_data\train"," label_file_list:D:\TOOLSSS\py3.11\Lib\site-packages\train_data\det\val.txt"
These parameters don't seem to be related to the data type？
Maybe you mean that the original configuration file had some error？
![2024-09-18 153618](https://github.com/user-attachments/assets/153a9cc1-d7d6-4d31-a615-f005878ea5fe)

**Reply by UserWangZz** [reply]
could you try https://github.com/PaddlePaddle/PaddleOCR/blob/main/configs/det/ch_PP-OCRv4/ch_PP-OCRv4_det_student.yml
this config for finetune pp-ocrv4-det_mobile model.
and the ch_PP-OCRv4_det_teacher.yml is a server model for a bigger model size and higher accuracy.
